cue/ui/gen.cue

```diff
@@ -93,3 +125,54 @@ VizLegendOptions: {
 VizTooltipOptions: {
 	mode: TooltipDisplayMode
 } @cuetsy(targetType="interface")
+// TODO copy back to appropriate place
+SingleStatBaseOptions: {
+	OptionsWithTextFormatting
+	reduceOptions: ReduceDataOptions
+	orientation: VizOrientation
+} @cuetsy(targetType="interface")
+// TODO copy back to appropriate place
+ReduceDataOptions: {
+	// If true show each row value
+	values?: bool
+	// if showing all values limit
+	limit?: number
+	// When !values, pick one value for the whole field
+	calcs: [...string]
+	// Which fields to show.  By default this is only numeric fields
+	fields?: string
+} @cuetsy(targetType="interface")
+// TODO copy back to appropriate place
+VizOrientation: "auto" | "vertical" | "horizontal" @cuetsy(targetType="enum")
+// TODO copy back to appropriate place
+OptionsWithTooltip: {
+	// FIXME this field is non-optional in the corresponding TS type
+	tooltip?: VizTooltipOptions
+} @cuetsy(targetType="interface")
+// TODO copy back to appropriate place
+OptionsWithLegend: {
+	// FIXME this field is non-optional in the corresponding TS type
```

Comment:
This FIXME seems the best thing to do for now, @ryantxu . My guess is that, until we've actually got cuetsy doing the conversions to TS (including those defaults), the conversation may be a bit too abstract.

Fortunately, that should be Coming Soon™ once this is in :)